pos_cache_user_restriction/tests/test_pos_cache_user_restriction.py

```diff
@@ -0,0 +1,55 @@
+from odoo.tests.common import SavepointCase
+
+
+class TestUserRestriction(SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env = cls.env(context=dict(
+            cls.env.context,
+            tracking_disable=True,
+            no_reset_password=True,
+        ))
+        cls.pos_user = cls.env['res.users'].create({
+            'login': 'pos_user',
+            'name': 'pos_user',
+            'groups_id': [(6, 0, [cls.env.ref('point_of_sale.group_pos_user').id])]
+        })
+        cls.pos_user_assigned_pos = cls.env['res.users'].create({
+            'login': 'pos_user_assigned_pos',
+            'name': 'pos_user_assigned_pos',
+            'groups_id': [(6, 0, [cls.env.ref(
+                'pos_user_restriction.group_assigned_points_of_sale_user').id])]
+        })
+        cls.pos_config_main = cls.env.ref('point_of_sale.pos_config_main')
+        cls.pos_config_model = cls.env['pos.config']
+
+    def test_access_pos(self):
+        # assigned_user_ids is not set: both users can read
+        pos_configs = self.pos_config_model.sudo(self.pos_user.id).search([])
+        self.assertTrue(pos_configs)
+        pos_configs = self.pos_config_model.sudo(
+            self.pos_user_assigned_pos.id).search([])
+        self.assertTrue(pos_configs)
+
+        self.pos_config_main.assigned_user_ids = [
+            (6, 0, [self.pos_user_assigned_pos.id])]
+        # assigned_user_ids is set with pos_user_assigned_pos: both users can read
+        pos_configs = self.pos_config_model.sudo(self.pos_user.id).search([])
+        self.assertTrue(pos_configs)
+        pos_configs = self.pos_config_model.sudo(
+            self.pos_user_assigned_pos.id).search([])
+        self.assertTrue(pos_configs)
+
+        self.pos_config_main.assigned_user_ids = [
+            (6, 0, [self.pos_user.id])]
+        # assigned_user_ids is set with pos_user: only pos_user can read
+        pos_configs = self.pos_config_model.sudo(self.pos_user.id).search([])
+        self.assertTrue(pos_configs)
+        pos_configs = self.pos_config_model.sudo(
+            self.pos_user_assigned_pos.id).search([])
+
+        # TODO, fixme
+        # this test is failing, if Odoo pos_restaurant is installed
+        # self.assertFalse(pos_configs)
```

Comment:
I'm wondering if this will create issues if we ever add a module depending on `pos_restaurant` :sweat_smile: 
Is it really failing or just a leftover TODO comment?

Reply:
@eLBati ?

Reply:
@ivantodorovich i don't think so. IMO pos_restaurant makes fail the test because it is creating a second pos.config (in the demo data). In fact, any modules adding a ``pos.config`` will make this test failing. 

I think this test should be rewrited, by writing. (maybe with a domain in the ``search([])``)

Reply:
Oh yes, indeed.
@GSLabIt can you take care of it?

instead of testing with a domain that'll get you all the possible pos.config, make sure to test with a domain or with specific config ids

Reply:
@ivantodorovich @legalsylvain test removed as it is not needed in this module. Please refer to @eLBati and https://github.com/OCA/pos/tree/14.0/pos_user_restriction for further changes.

Reply:
I think you should remove these tests files from the module. They seem to be a copy of pos_user_restriction's tests, or are you testing something new on purpose?

Reply:
TBH I think in this case we don't need tests.
But I see your point. If you want to run exactly the same tests, though.. rather than copy/pasting it I would reuse the existing test suite

```py
from odoo.addons.pos_user_restriction.tests.test_pos_user_restriction import TestUserRestriction


class TestPosCacheUserRestriction(TestUserRestriction):
    """Run pos_user_restriction's test suite with pos_cache installed"""
    pass
```

Alternatively, what about changing pos_user_restriction's tests so that they run at `post_install` instead of `at_install` ?
https://github.com/odoo/odoo/blob/e935cf9a51004aa3dff5699ad4da09c46c09d42d/addons/website/tests/test_ui.py#L8

By doing that, pos_user_restriction's tests would ran after all modules are installed. So it would test the compatibility with pos_cache. https://www.odoo.com/documentation/12.0/fr/developer/reference/testing.html#odoo.tests.common.post_install

Reply:
> Alternatively, what about changing pos_user_restriction's tests so that they run at `post_install` instead of `at_install` ?
> https://github.com/odoo/odoo/blob/e935cf9a51004aa3dff5699ad4da09c46c09d42d/addons/website/tests/test_ui.py#L8
> 
> By doing that, pos_user_restriction's tests would ran after all modules are installed. So it would test the compatibility with pos_cache. https://www.odoo.com/documentation/12.0/fr/developer/reference/testing.html#odoo.tests.common.post_install

Let's go with this, as same tests will be needed for: https://github.com/OCA/pos/issues/696. I've removed test from this module